### ISSUE TITLE
An question about "accuracy of different clsf"

### ISSUE THREAD

**Issue by zwxandy** [klrc/RACNN-pytorch#3]
Hi, @klrc
I just started learning RA-CNN, and I have a question for you. Here is your result on the testset:

> [2019-12-31 20:06:50]    :: Testing on test set ...
[2019-12-31 20:07:10]           Accuracy clsf-0@top-1 (201/725) = 79.95050%
[2019-12-31 20:07:10]           Accuracy clsf-0@top-5 (201/725) = 94.61634%
[2019-12-31 20:07:10]           Accuracy clsf-1@top-1 (201/725) = 74.25743%
[2019-12-31 20:07:10]           Accuracy clsf-1@top-5 (201/725) = 91.39851%
[2019-12-31 20:07:10]           Accuracy clsf-2@top-1 (201/725) = 74.62871%
[2019-12-31 20:07:10]           Accuracy clsf-2@top-5 (201/725) = 90.71782%

I wonder why the accuracy of clsf-0, clsf-1, clsf-2 decreases in order? Shouldn't clsf-2 be a more fine-grained region compared with clsf-1 and clsf-0? 
I got the same results on my datasets. If you can help me with this question, I would appreciate it very much! Thank you!

**Comment by klrc** [comment]
Hi, @zwxandy 
In fact, I have the same issue during the re-implementation of this paper. Also, it seems that during training, the 'zoom' effect fades away eventually. (see #1) 
It has been a long time since I implemented this, I'm sorry that i can't test this further.
Anyway, I think its an issue with some hyperparameters in margin loss, causing the model failing to detect by zooming. Maybe some tricks like increasing the margin loss weight would solve this.

here is another implementation: https://github.com/jeong-tae/RACNN-pytorch, maybe you can try this.
Good Luck!

**Comment by zwxandy** [comment]
> Hi, @zwxandy In fact, I have the same issue during the re-implementation of this paper. Also, it seems that during training, the 'zoom' effect fades away eventually. (see #1) It has been a long time since I implemented this, I'm sorry that i can't test this further. Anyway, I think its an issue with some hyperparameters in margin loss, causing the model failing to detect by zooming. Maybe some tricks like increasing the margin loss weight would solve this.
> 
> here is another implementation: https://github.com/jeong-tae/RACNN-pytorch, maybe you can try this. Good Luck!

Got it, thank you very much! @klrc 
I think the extent of such a phenomenon is also related to the dataset. I still wanna explore the reason why "the model fails to detect by zooming", and I will make some attempts in my spare time. If I have any discoveries, I'd love to share with you.
Thanks again!
